TempleEight/docs/guide/authentication.md

```diff
@@ -3,3 +3,186 @@ id: authentication
 title: Authentication
 sidebar_label: Authentication
 ---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+A crucial component of most systems is its authentication system, namely validating that the user of a system is who they claim to be.
+At Temple, we baked this directly into the generated architecture, so that you don't have to worry about getting the details right.
+In this guide, we'll show you how to add authentication to your project and how this affects the system architecture, using the example project from the [Getting Started](../getting-started) guide.
+
+## Adding Authentication to Your Project
+The [Getting Started](../getting-started) guide concluded with the following example Templefile:
+
+```
+ExampleProject: project {
+  #language(go);
+  #database(postgres);
+  #provider(dockerCompose);
+}
+
+ExampleService: service {
+  foo: string;
+  bar: int;
+}
+```
+
+To add authentication to this, we need to add two things:
+
+1. **A project-level auth method metdata**
+  * This indicates how users will log in. Currently we only support `email` authentication, but we will be expanding this in the future.
+2. **A service-level auth metadata**
+  * This indicates one or many services which store metadata about your users, in addition to their authentication details.
+
+We're going to apply these two modifications to our example Templefile by introducing a new service called `ExampleUser`, which will additionally store the user's name:
+
+```templefile {5,8-11}
```

Comment:
Does this work????

Reply:
It doesn't highlight Templefile, but it does the specific line highlighting